weather.go

```diff
@@ -0,0 +1,111 @@
+package weather
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"math"
+	"net/http"
+	"os"
+	"strings"
+)
+
+type apiResponse struct {
+	Name    string
+	Weather []struct {
+		Main string
+	}
+	Main struct {
+		Temp float64
+	}
+}
+
+type Conditions struct {
+	City               string
+	OneWord            string
+	TemperatureCelsius float64
+}
+
+func RunCLI(args []string) {
+
+	token := os.Getenv("WEATHERAPP_TOKEN")
+	if token == "" {
+		fmt.Fprintf(os.Stderr, "please set env variable, WEATHERAPP_TOKEN \n")
+		os.Exit(2)
+	}
+
+	request, err := Request(args, token)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "problem setting url', %v", err)
+		os.Exit(2)
+	}
+
+	response, err := Response(request)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "an error has occured, %v", err)
```

Comment:
No need to say "an error has occured"—that's implied, isn't it? We can just print the error directly.

```suggestion
		fmt.Fprintln(os.Stderr, err)
```